### PR TITLE
Added features to support the OMI-Script-Provider.

### PR DESCRIPTION
**Pull request by EMumau**
These changes are needed to support the OMI-Script-Provider.

getopt.c:
    added option arguments for command line args

configure:
    added registration for omireg.conf

Base_OMI.data:
    added file placement for omireg.conf

omireg.cpp:
    added a configuration file to allow future additions for new scripting languages without core changes
    added opening processing for script providers
    added new arguments to the output files to support scripting providers

provmgr.c:
    added an alternative processing case to handle scripting providers

provereg.[ch]:
    added the members "interpreter" and "startup" to ProvRegEntry

regfile.[ch]:
    added the members "interpreter" and "startup" to RegFile

messages.[ch]:
    replaced "libraryName" and "instanceLifetimeContext" with "regEntry" in the RequestMsg structure.

agent.c:
    support for the changes to the content of the RequestMsg data structure

agentmgr.c:
    support for the changes to the content of the RequestMsg data structure

### CODE REVIEW

File: Unix/configure [hunk 1]
```diff
@@ -307,10 +302,10 @@ do
             enable_ulinux=1
         fi
 
-        if [ `uname -m` != "x86_64" ]; then
-            echo "Note: Will NOT build for OpenSSL 1.1.0 on 32-bit platforms ..."
+	if [ `uname -m` != "x86_64" ]; then
```
Comment (alinbalutoiu) [inline Unix/configure]: nit: unnecessary changes, you replaced spaces with tabs here

File: Unix/configure [hunk 2]
```diff
@@ -292,11 +292,6 @@ do
         fi
 
         enable_ulinux=1
-
-        if [ `uname -m` != "x86_64" ]; then
-            echo "Note: Will NOT build for OpenSSL 1.1.0 on 32-bit platforms ..."
-            disable_ssl_1_1_0=1
-        fi
```
Comment (alinbalutoiu) [inline Unix/configure]: Why was this removed?

File: Unix/installbuilder/datafiles/Base_OMI.data [hunk 3]
```diff
@@ -27,26 +27,25 @@ PAM_CONF_DIR: '/etc/pam.d'
 /opt/omi/bin/omireg;                          bin/omireg;                            755; root; ${{ROOT_GROUP_NAME}}
 /opt/omi/bin/omigen;                          bin/omigen;                            755; root; ${{ROOT_GROUP_NAME}}
 /opt/omi/bin/omiconfigeditor;                 bin/omiconfigeditor;                   755; root; ${{ROOT_GROUP_NAME}}
-/opt/omi/lib/libmi.${{SHLIB_EXT}};            lib/libmi.${{SHLIB_EXT}};              755; root; ${{ROOT_GROUP_NAME}}
+/opt/omi/lib/libmi.${{SHLIB_EXT}};            lib/libmi.${{SHLIB_EXT}};           755; root; ${{ROOT_GROUP_NAME}}
```
Comment (alinbalutoiu) [inline Unix/installbuilder/datafiles/Base_OMI.data]: nit: unnecessary change

File: Unix/installbuilder/datafiles/Base_OMI.data [hunk 4]
```diff
@@ -27,26 +27,25 @@ PAM_CONF_DIR: '/etc/pam.d'
...
 /etc/opt/omi/conf/omiserver.conf;             ../installbuilder/conf/omiserver.conf; 444; root; sys; conffile
 /etc/opt/omi/conf/omigen.conf;                ../etc/omigen.conf;                    444; root; sys
+/etc/opt/omi/conf/omireg.conf;                ../etc/omireg.conf;                    444; root; sys
 /etc/opt/omi/conf/omiregister/root-omi/omiidentify.reg; ../etc/omiregister/root-omi/omiidentify.reg; 755; root; ${{ROOT_GROUP_NAME}}
 
 %Directories
 /opt;                                    755; root; ${{ROOT_GROUP_NAME}}; sysdir
 /opt/omi;                                755; root; ${{ROOT_GROUP_NAME}}
 /opt/omi/bin;                            755; root; ${{ROOT_GROUP_NAME}}
-/opt/omi/bin/support;                    755; root; ${{ROOT_GROUP_NAME}}
```
Comment (alinbalutoiu) [inline Unix/installbuilder/datafiles/Base_OMI.data]: It looks like you reverted some changes from commit: https://github.com/Microsoft/omi/commit/dee95dcf574dc417080053fa780b869e7ce4db9d
Was this intended?

**Comment by EMumau** [comment]
I made changes for the issues that Alin commented on.
